### PR TITLE
[Dev] Add Qwen3-VL support with Megatron-FSDP

### PR DESCRIPTION
**Pull request by xuwchen**
# What does this PR do ?
main PR: [#2841](https://github.com/NVIDIA/Megatron-LM/pull/2841) 

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact the @megatron-oncall.

## Contribution process

```mermaid
flowchart LR
    A[Pre-checks] --> B[PR Tests]
    subgraph Code Review/Approval
        C1[Expert Review] --> C2[Final Review]
    end
    B --> C1
    C2 --> D[Merge]
```

### Pre-checks

- [ ] I want this PR in a versioned release and have added the appropriate Milestone (e.g., `Core 0.8`)
- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

The following process is enforced via the CODEOWNERS file for changes into `megatron/core`. For changes outside of `megatron/core`, it is up to the PR author whether or not to tag the Final Reviewer team.

<details>
<summary>For MRs into `main` branch</summary>

Feel free to message or comment the @megatron-oncall to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

#### (Step 2): Collect the expert reviewers reviews

1. Attach the `Expert Review` label when your PR is ready for review.
2. GitHub auto-assigns expert reviewers based on your changes. They will get notified and pick up your PR soon.

:warning: Only proceed to the next step once all reviewers have approved, merge-conflict are resolved and the CI is passing.  
Final Review might get declined if these requirements are not fulfilled.

#### (Step 3): Final Review

1. Add `Final Review` label
2. GitHub auto-assigns final reviewers based on your changes. They will get notified and pick up your PR soon.

#### (Optional Step 4): Cherry-pick into release branch

If this PR also needs to be merged into `core_r*` release branches, after this PR has been merged, select `Cherry-pick` to open a new PR into the release branch.

</details>

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### Merging your PR

Any member of [core-adlr](https://github.com/orgs/teams/NVIDIA/core-adlr) and [`core-nemo`](https://github.com/orgs/teams/NVIDIA/core-nemo) will be able to merge your PR.

### CODE REVIEW

File: megatron/core/transformer/fsdp_dtensor_checkpoint.py [hunk 1]
```diff
@@ -196,7 +197,8 @@ def handle_swiglu_in_state_dict(model, model_state_dict, optimizer_state_dict):
     assert HAVE_MEGATRON_FSDP, "This function requires Megatron-FSDP to be installed."
 
     # Extract num_experts from model config for expert parameter processing
-    num_experts = model.config.num_moe_experts if hasattr(model, 'config') else None
+    model_config = get_model_config(model)
```
Comment (BestJuly) [inline megatron/core/transformer/fsdp_dtensor_checkpoint.py]: Would this cause an error if the model does not have config? Suggest to use `model_config = get_model_config(model, allow_none=True)` here, and modify the API of `get_model_config` (default value can be False to match with original code logic).

Comment (xuwchen) [reply megatron/core/transformer/fsdp_dtensor_checkpoint.py]: This function is only used in the Megatron-FSDP checkpoint flow, where models are expected to have a `config` attribute. The `RuntimeError` from `get_attr_wrapped_model` provides a useful early failure signal if something is misconfigured. Prefer keeping the fail-fast behavior here.

**Review by shjwudp — APPROVED** [review]
LGTM, thanks

File: megatron/core/transformer/fsdp_dtensor_checkpoint.py (hunk 1 above)
Comment (yanring) [reply megatron/core/transformer/fsdp_dtensor_checkpoint.py]: What if a torch native model is passed in instead of a Megatron model?

**Comment by yanring** [comment]
/ok to test [6b1133e](https://github.com/NVIDIA/Megatron-LM/pull/2842/commits/6b1133e7765d9145901a779bfd745358100e21f6)

**Comment by ko3n1g** [comment]
🔄 Merge queue validation started!

You can track the progress here: https://github.com/NVIDIA/Megatron-LM/actions/runs/21972774149

**Comment by ko3n1g** [comment]
🔄 Merge queue validation started!

You can track the progress here: https://github.com/NVIDIA/Megatron-LM/actions/runs/21975008659
